pypdf/_writer.py

```diff
@@ -2100,7 +2101,11 @@ def removeLinks(self) -> None:  # deprecated
         return self.remove_links()
 
     def remove_annotations(
-        self, subtypes: Optional[Union[AnnotationSubtype, Iterable[AnnotationSubtype]]]
+        self,
+        subtypes: Optional[
+            Union[AnnotationSubtype, Iterable[AnnotationSubtype]]
+        ] = None,
+        annotation_filter_function: Optional[Callable] = None,
```

Comment:
Just an idea: We could call the parameter `exclude`. That (in combination with the type annotation and the docs) seems pretty clear to me. Whenever it evaluates to `True`, the entry is excluded. That would inverse the logic.

I'm not happy with `annotation_filter_function` as it's pretty lengthy + I never know if we "filter out" (remove) or "filter to keep".

We could then make the default the constant-False function (which would not remove anything). This would mean the annotation would become simpler as the "Optional" could be removed.

Then the meaning of subtypes in combination with the new parameter has to be clear. I think it would be best if the both are independent filter functions. If both are set, we can create a new function that combines both internally. That means we don't have to ignore one parameter. And we don't have to log anything.

Reply:
Hm, I don't really like `exclude`, because it feels more natural to `return True` to keep something. Maybe `include` or `filter`?

> We could then make the default the constant-False function (which would not remove anything). This would mean the annotation would become simpler as the "Optional" could be removed.

I'm not sure I get this. I wanted to set the `default_annotation_filter_function` as the `annotation_filter_function` by default, too, without `Optional[Callable] = None`. But `mypy` didn't let me iirc.

> I think it would be best if the both are independent filter functions. If both are set, we can create a new function that combines both internally. That means we don't have to ignore one parameter. And we don't have to log anything.

Agreed.